examples/plot_learn_from_imbalanced_dataset.py

```diff
@@ -0,0 +1,276 @@
+    global df_scores
+    df_scores = pd.concat([df_scores, clf_score], axis=0).round(decimals=3)
+
+
+# Let's define an empty dataframe to store the results
+df_scores = pd.DataFrame()
+
+###############################################################################
+# Dummy baseline
+# ..............
+#
+# Before to train a real machine learning model, we can store the results
+# obtained with our :class:`sklearn.dummy.DummyClassifier`.
+
+evaluate_classifier(dummy_clf)
+df_scores
+
+###############################################################################
+# Linear classifier baseline
+# ..........................
+#
+# We will create a machine learning pipeline using a
+# :class:`sklearn.linear_model.LogisticRegression` classifier. In this regard,
+# we will need to one-hot encode the categorical columns and standardized the
+# numerical columns before to inject the data into the
+# :class:`sklearn.linear_model.LogisticRegression` classifier.
+#
+# First, we define our numerical and categorical pipelines.
+
+from sklearn.impute import SimpleImputer
+from sklearn.preprocessing import StandardScaler
+from sklearn.preprocessing import OneHotEncoder
+from sklearn.pipeline import make_pipeline
+
+num_pipe = make_pipeline(
+    StandardScaler(), SimpleImputer(strategy="mean", add_indicator=True)
+)
+cat_pipe = make_pipeline(
+    SimpleImputer(strategy="constant", fill_value="missing"),
+    OneHotEncoder(handle_unknown="ignore")
+)
+
+###############################################################################
+# Then, we can create a preprocessor which will dispatch the categorical
+# columns to the categorical pipeline and the numerical columns to the
+# numerical pipeline
+
+from pandas.api.types import CategoricalDtype
+from sklearn.compose import ColumnTransformer
+from sklearn.compose import make_column_selector as selector
+
+preprocessor_linear = ColumnTransformer(
+    [("num-pipe", num_pipe, selector(dtype_include=np.number)),
+     ("cat-pipe", cat_pipe, selector(dtype_include=CategoricalDtype))],
+    n_jobs=2
+)
+
+###############################################################################
+# Finally, we connect our preprocessor with our `LogisticRegression`. We can
+# then evaluate our model.
+
+from sklearn.linear_model import LogisticRegression
+
+lr_clf = make_pipeline(
+    preprocessor_linear, LogisticRegression(max_iter=1000)
+)
+evaluate_classifier(lr_clf)
+df_scores
+
+###############################################################################
+# We can see that our linear model is learning slightly better than our dummy
+# baseline. However, it is impacted by the class imbalance.
+#
+# We can verify that something similar is happening with a tree-based model
+# such as :class:`sklearn.ensemble.RandomForestClassifier`. With this type of
+# classifier, we will not need to scale the numerical data, and we will only
+# need to ordinal encode the categorical data.
```

Comment:
Because it will be inefficient in practise. You are going to create sparse columns which you will scan when seeking for the best splits. Ordinal encoding migh require 2 splits to isolate a specific categories but it will be really efficient.